File: prob217.ts
function containsDuplicate(nums: number[]): boolean {
  const set = new Set();
  for (let i = 0; i < nums.length; i++) {
    const val = nums[i];
    if (set.has(val)) {
      return true;
    } else {
      set.add(val);
    }
  }
  return false;
}

File: prob217.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        hashset = set()
        for num in nums:
            if num in hashset:
                return True
            else:
                hashset.add(num)
        return False


sol = Solution()
nums = [1, 2, 3, 5]
print(sol.containsDuplicate(nums))
